### PR TITLE
Sunbird CDN configuration

### PR DESCRIPTION
**Pull request by surabhi-mahawar**
**Jira ticket:** [SB-31158](https://project-sunbird.atlassian.net/browse/SB-31158)

**Description:** This PR includes the environment variables required for sunbird cloud agnostic tool. We have added conditions to use this tool or other services based on the configuration. This tool provides option to use AWS/Azure etc which will also be handled via sunbird confirguation variables **SUNBIRD_CLOUD_MEDIA_STORAGE_TYPE**.

We intend to store all the media files(Image/Audio/Video/Document) being used in UCI assessment bot. Also use this to display media files as well.

If these configuration is not set, it will start the service but will just show error message in the service logs and will not respond to that particular media related message.

### CODE REVIEW

**Comment by surabhi-mahawar** [comment]
@keshavprasadms 
Can you please merge this PR.

**Comment by keshavprasadms** [comment]
@surabhi-mahawar 
- Please add a details description on what this change is about
- Jira link references
- Documentation link if any
- What do you intend to store in these buckets / containers
- What would happen if CDN / media storage is not configured for an environment
- What about other CSP's implementation

**Comment by surabhi-mahawar** [comment]
@keshavprasadms 
I have updated the description of the PR with jira ticket and the required information you mentioned. 
We have added these environment variables for the sunbird's cloud agnostic tool only.

File: ansible/roles/stack-sunbird/templates/transformer.env [hunk 1]
```diff
@@ -72,6 +72,14 @@ POSTHOG_EVENT_ENABLED=FALSE
 
 #Azure Config
 AZURE_BLOB_STORE_CONTAINER={{sunbird_azure_uci_container_name | default('uci-' + env )}}
-AZURE_BLOB_STORE_ACCOUNT_KEY={{sunbird_private_storage_account_name}}
-AZURE_BLOB_STORE_ACCOUNT_NAME={{sunbird_private_storage_account_key}}
-SELECTED_FILE_CDN=azure
\ No newline at end of file
+AZURE_BLOB_STORE_ACCOUNT_NAME={{sunbird_private_storage_account_name}}
```
Comment (keshavprasadms) [inline ansible/roles/stack-sunbird/templates/transformer.env:75]: @surabhi-mahawar Are these variable still required? Below is another variable which has same contents
`SUNBIRD_CLOUD_MEDIA_STORAGE_KEY`

**Comment by keshavprasadms** [comment]
@surabhi-mahawar This PR is supporting only Azure cloud right? In the description, I can see AWS support also, but I am unable to find the corresponding AWS cloud variables in the PR.

File: ansible/roles/stack-sunbird/templates/transformer.env (hunk 1 above)
Comment (surabhi-mahawar) [reply ansible/roles/stack-sunbird/templates/transformer.env]: No this is not required not, I have removed these variables from the files

**Comment by surabhi-mahawar** [comment]
> @surabhi-mahawar This PR is supporting only Azure cloud right? In the description, I can see AWS support also, but I am unable to find the corresponding AWS cloud variables in the PR.

@keshavprasadms The sunbird cloud agnostic tool should handle this according to the **SUNBIRD_CLOUD_MEDIA_STORAGE_TYPE** mentioned, if this changes to AWS, the credentials for it should be added in **SUNBIRD_CLOUD_MEDIA_STORAGE_KEY, SUNBIRD_CLOUD_MEDIA_STORAGE_SECRET** as well. 
It should change for everyone right, not just for one instance. 
It should be either Azure or AWS. 

Or can there be a case where different instances can have different CDNs ?

**Comment by keshavprasadms** [comment]
@surabhi-mahawar Please can you connect with @santhosh-tg for this? He is driving a change across all BB's to standardize the variable names and the naming convention, how to support multiple clouds etc. A brief discussion with him would help you get clarity and take this PR forward. Would be good if you can please setup a call and discuss with the team - @santhosh-tg @gandham-santhosh

**Comment by surabhi-mahawar** [comment]
> @surabhi-mahawar Please can you connect with @santhosh-tg for this? He is driving a change across all BB's to standardize the variable names and the naming convention, how to support multiple clouds etc. A brief discussion with him would help you get clarity and take this PR forward. Would be good if you can please setup a call and discuss with the team - @santhosh-tg @gandham-santhosh

@keshavprasadms 
Can you please send us the email IDs for @santhosh-tg @gandham-santhosh, so we can connect with them.

**Comment by keshavprasadms** [comment]
I don't think adding email id's here is a good idea due to privacy concerns. As @ChakshuGautam is already on Slack, you can easily get in touch with the folks. These are the slack handlers
- `@Santhosh Gandham`
- `@Santhosh Kumar Sreedharan`

**Comment by surabhi-mahawar** [comment]
@keshavprasadms 
I have made the changes in the PR as per the variables mentioned in the sheet & discussion with Santhosh Kumar Sreedharan.
Can you please review this PR.

CC: @ChakshuGautam

File: ansible/roles/stack-sunbird/templates/inbound.env [hunk 2]
```diff
@@ -53,14 +53,16 @@ REDIS_PASS={{sunbird_redis_pass | default('')}}
...
-SELECTED_FILE_CDN=azure
-
 #Netcore
 NETCORE_WHATSAPP_AUTH_TOKEN={{uci_netcore_whatsapp_token}}
 NETCORE_WHATSAPP_SOURCE={{uci_netcore_whatsapp_source}}
 NETCORE_WHATSAPP_URI={{uci_netcore_whatsapp_uri | default('https://waapi.pepipost.com/api/v2/')}}
 
+#Sunbird CDN Configuration
+SUNBIRD_CLOUD_MEDIA_STORAGE_TYPE={{cloud_service_provider}}
+SUNBIRD_CLOUD_MEDIA_STORAGE_KEY={{cloud_public_storage_accountname}}
```
Comment (santhosh-tg) [inline ansible/roles/stack-sunbird/templates/inbound.env:63]: @surabhi-mahawar  I think the storage account is private. Please check and confirm. If yes, change the accountname and secret vars accordingly.

Comment (surabhi-mahawar) [reply ansible/roles/stack-sunbird/templates/inbound.env]: @santhosh-tg 
I have changed the storage account variables to private. 
Please review it.

**Review by santhosh-tg — APPROVED** [review]

**Pull request merged**